docs/core/versions/binding.md

```diff
@@ -0,0 +1,170 @@
+
+Consider an application requiring 2.0.4, `dotnet` would search and bind to a runtime in this order:
+
+- 2.0.* (highest patch version; ensure that it satisfies 2.0.4 as a lower bound)
+- 2.* (highest minor version)
+
+If neither is found, you see an error message similar to the following example:
+
+> This application requires .NET Core 2.0.4. Please install .NET Core 2.0.4 or a higher compatible version.
+> Please see https://aka.ms/install-dotnet-core to learn about .NET Core installation and runtime versioning.
+
+A few usage examples demonstrate the behavior:
+
+- 2.0.4 is required. 2.0.5 is the highest patch version available. 2.0.5 is used.
+- 2.0.4 is required. No 2.0.* versions are installed. 1.1.1 is the highest runtime installed. An error message is printed.
+- 2.0.4 is required. No 2.0.* versions are installed. 2.2.2 is the highest 2.x runtime version installed. 2.2.2 is used.
+- 2.0.4 is required. No 2.x versions are installed. 3.0.0 is installed. An error message is printed.
+
+Minor version roll-forward has one side-effect that may affect end users. Consider the following scenario:
+
+- 2.0.4 is required. No 2.0.* versions are installed. 2.2.2 is installed. 2.2.2 is used.
+- 2.0.5 is later installed. 2.0.5 will be used for subsequent application launches, not 2.2.2.
+- It's possible that 2.0.5 and 2.2.2 might behave differently, particularly for scenarios like serializing binary data.
+
+NuGet dependencies are treated differently. An application might be built for .NET Core 2.0 but run on .NET Core 2.1 due to this algorithm. There may be later versions of an application's NuGet dependencies that target .NET Core 2.1. These later versions aren't considered or installed. All NuGet dependencies are resolved as part of the publish operation and reside as assemblies (.dlls) in a flat directory structure. These assemblies should run on .NET Core 2.1 due to its compatibility promise for existing applications and binaries.
+
+Occasionally, ASP.NET packages are deployed via a web host rather than with a published application. In those cases, the web host should correctly configure their environments based on published guidance such that all supported ASP.NET applications run correctly.
+
+Developers should remember the following rules:
+
+- Runtimes are searched for in the following order (the first found is loaded):
+  - Highest patch version (check that the minimum runtime path version is satisfied, otherwise error)
+  - Highest minor version
+- If an acceptable runtime is not found, the application does not run and an error message explains why.
+
+## Publish a self-contained application
+
+You can publish an application as a **self-contained distribution**. This approach includes .NET Core. Self-contained distributions do not have a dependency on runtime environments. Runtime binding occurs at publishing time, not run-time.
```

Comment:
link to https://docs.microsoft.com/en-us/dotnet/core/deploying/#self-contained-deployments-scd for the definition of self-contained and the other anchor framework-dependent apps. Also, do we need to link from that topic back here for learning about version binding at publishing time?

Reply:
in second sentence: "includes NET Core runtime."

Drop the third sentence. It is incorrect. The fourth is the point. If you like, move the third to fourth and restate to be very clear you mean it does not depend on a framework that is installed on the machine.